core/web/api/links.py

```diff
@@ -66,3 +66,51 @@ def multiupdate(self):
             updated.append(link.id)
 
         return render({"updated": updated})
+
+    @route("/", methods=['POST'])
+    @requires_permissions('write')
+    def new(self):
+        """Create a new link
+
+        Create a new link from the JSON object passed in the ``POST`` data.
+
+        :<json object params: JSON object containing object ids to link
+        """
+
+        type_map = {
+            'observable': observables.Observable,
+            'entity': entities.Entity,
+            'indicator': indicators.Indicator
+        }
+
+        mandatory_params = ['type_src', 'type_dst', 'link_src', 'link_dst']
+        params = request.json
+
+        if all(key in params for key in mandatory_params):
+
+            type_src = params.get('type_src')
+            type_dst = params.get('type_dst')
+            src_object_class = type_map.get(type_src, None)
+            dst_object_class = type_map.get(type_dst, None)
+
+            if src_object_class and dst_object_class:
```

Comment:
To make code flow easier to read and have nicer indentation, you can use a guard statement here:

```python
# guard statement
if not all(key in params for key in mandatory_params):
    abort (400)

type_src = params.get('type_src')
type_dst = params.get('type_dst')
src_object_class = type_map.get(type_src, None)
dst_object_class = type_map.get(type_dst, None)

# guard statement
if not src_object_class or not dst_object_class:
  abort(404)

# proceed as normal; you don't even need else clauses
if params.get('first_seen', None)...
    # bla bla bla
else:
    # bla bla bla
return render({'link': link})
```